search/controller/controller/settings.py

```diff
@@ -0,0 +1,125 @@
+"""
+Django settings for controller project.
+
+Generated by 'django-admin startproject' using Django 2.2.
+
+For more information on this file, see
+https://docs.djangoproject.com/en/2.2/topics/settings/
+
+For the full list of settings and their values, see
+https://docs.djangoproject.com/en/2.2/ref/settings/
+"""
+
+import os
+
+# Build paths inside the project like this: os.path.join(BASE_DIR, ...)
+BASE_DIR = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
+
+
+# Quick-start development settings - unsuitable for production
+# See https://docs.djangoproject.com/en/2.2/howto/deployment/checklist/
+
+# SECURITY WARNING: keep the secret key used in production secret!
+SECRET_KEY = "sp5yk-!1ckot-kb!e23lqiw45wg4yv#7!z@p-*6hv!2bfk4mx_"
```

Comment:
I assume this isn't the one used in the production server, as it is now public? We can use docker secrets to store a new secret key, assuming it is supported by the current version of docker installed, but I've not used it before.

Reply:
Do we set it as an environment variable in the Dockerfile? That would be visible as well from repo as well. Or we set it using docker command separately.

Reply:
Docker secrets is a functionality of docker. See [here](https://docs.docker.com/engine/swarm/secrets/). Secrets can either be loaded from file as part of a docker-compose operation (but that file would not be committed to git) or can be created with `docker secret create`.

This is the Docker Compose syntax for top-level syntax definitions:
```
secrets:
  my_first_secret:
    file: ./secret_data
  my_second_secret:
    external: true
```
You then make the secret available to individual containers with:
```
my_container:
  secrets:
    - my_first_secret
```
Then within the container each secret is a file in `/run/secrets`, e.g. you can cat `/run/secrets/my_first_secret`.

The simpler, less secure alternative is to set secrets in a .env file which is used as environment for docker-compose and can be passed to individual containers' environments, as here: https://github.sydney.edu.au/informatics/pipe-274-media-downloader/blob/master/.env
https://github.sydney.edu.au/informatics/pipe-274-media-downloader/blob/8d96ab1b77299c75177fc8b123d4c57fce9d4dc0/docker-compose.yml#L18-L21

Reply:
Having this issue, it seems the docker-compose version on vm may not support secrets.

[root@wdidpa00000 search]# docker-compose build
ERROR: The Compose file './docker-compose.yml' is invalid because:
Invalid top-level property "secrets". Valid top-level sections for this Compose file are: version, services, networks, volumes, and extensions starting with "x-".

You might be seeing this error because you're using the wrong Compose file version. Either specify a supported version (e.g "2.2" or "3.3") and place your service definitions under the `services` key, or omit the `version` key and place your service definitions at the root of the file to use version 1.
For more on the Compose file format versions, see https://docs.docker.com/compose/compose-file/
Unsupported config option for services.controller: 'secrets'

Reply:
The docker-compose file version we are using should be 3.1, which is aligned with Docker 1.13.1